### ISSUE TITLE
Assignment #3

### ISSUE THREAD

**Issue by itscoe** [itscoe/CS390#2]
**Describe the solution you'd like**
This iteration focuses on enhancing student interactions.  Add the capability for students to add a course, delete a course, check their current schedule, and display their course history.  Create a GUI that implements these transactions as given in the Group Project Specification. 

**Additional context**
On the due date each group will be given 5-10 minutes to demo and explain their code.  This process called a code review and is undertaken at important stages (or check-points) in the development process.  Groups will not be turning in their code; instead they will be showing and explaining how it works at this point in

**Comment by dswetlik** [comment]
I have figured out the error of my ways and have come forward to confess that I have made a stupid mistake in the reading of the files. It took a quick read-over of the specification document and actually using my brain to figure out the layout of the input files that he wanted and how to correctly read them in. I have been making an easy slope into a monumental mountain.

*** EDIT ***
To make the issue comprehensive, in the specification documents, I was reading
`<element-name:element-size>` 
to be
`<element-name:element-max-size>`
In other words, the fix is as simple as a loop of `file.Read() until x == `element-size`, rather than the current splitting the whole line into different strings based on a space char (' ').
*** END OF EDIT ***

Given enough time tonight/tomorrow, this issue should be fixed by Friday, Oct. 2.

**Comment by itscoe** [comment]
Awesome. I also added the add/drop functionality and current schedule, so about 75% of it is done now

**Comment by dswetlik** [comment]
@itscoe, just a quick question on your code. In this chunk

    public partial class StudentDashboard : Form
    {
        private Student current_user;
        private object sender;
        private EventArgs e;

        public StudentDashboard()
        {
            InitializeComponent();
            current_user = (Student)LogInScreen.current_user;
        }

        private void Form2_Load(object sender, EventArgs e)
        {
            sender = sender;
            e = e;
. . .

you set paramaters `object sender` and `EventArgs e` of `Form2_Load` to themselves. Is this intentional, or did you mean to do 
`
this.sender = sender;
this.e = e;
`

**Comment by itscoe** [comment]
@dswetlik The latter, but we also don't need that part anymore

**Comment by dswetlik** [comment]
I just remembered this as well, can someone update our UML to reflect what we need for the next assignment>

**Comment by Rumaldo** [comment]
I'll make sure to do that then, just reflect the add drop with
restrictions? or what exactly?

**Comment by dswetlik** [comment]
I made it slightly less convoluted, use helper methods "GetHistory___()" when accessing course history information, I'll probably make CourseHistory it's own class, but then I have to do other stuff to Course as well in case we want them to be switched over on a whim

**Comment by dswetlik** [comment]
Okay, new update. CourseHistory was made a list by mistake, one particular student should only have one history of courses. 

Accessing a Student's course history is somewhat simple. To access everything inside a student's history:
`Student student = _Logged In Student_;`
`Course courseHistory = student.GetCourseHistory();`
`for(int i = 0; i < courseHistory.GetHistoryNumCourses(); i++)`
     `Console.WriteLine(String.Format("{0} {1} {2} {3}", courseHistory.GetHistoryCourseIDs()[i], courseHistory.GetHistoryCourseCredits()[i],  courseHistory.GetHistoryTerms()[i],  courseHistory.GetHistoryGrades()[i]));`


Now that I think about it, it also should be pointed out that you might need to check for null values for a given student's courseHistory. Not every student that is created has a course history.

**Comment by dswetlik** [comment]
I updated the back-end code so it works more fluidly with the front-end. There remains an issue with the front-end, there is no way to toggle between the courses and course history sections, nothing was done with the buttons.

TL; DR: My back-end code works, screen pops up with correct course history after log-on, no way to see add-drop courses.

**Comment by itscoe** [comment]
The latest commit (c3fc8f5) has everything working (buttons on front-end, errors for enrolled in same course, same course different section, and scheduling conflicts, and toggling between add/drop and course history) that's required for this assignment. I'll try to get the tabs working for the per-semester view by class, but that's gravy.
